### DISCUSSION TITLE
TimeRampFunction is not a registered object

### DISCUSSION THREAD

**Discussion by AntNewm** [idaholab/moose#28365] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

I have successfully created and run a thermal hydraulics based script in the thermal hydraulics module of moose (projects/moose/modules/thermal_hydraulics/) to produce results etc. I would like to run it in my own application that I have created so that I can expand to use other modules. I made the application and enabled the modules I want in the makefile and then rebuilt the app. When I run the script it says 'A 'TimeRampFunction' is not a registered object'. 
Any help would be greatly appreciated!
(I am working on the HPC sawtooth interactive desktop.)

**Comment by GiudGiud** [comment]
Hello

Save all your work then `git clean -xfD` both moose/ and your application then recompile

Guillaume
